src/github_runner_image_builder/builder.py

```diff
@@ -0,0 +1,635 @@
+            ["/usr/bin/systemctl", "mask", APT_SVC], timeout=30
+        )  # nosec: B603
+        logger.info("systemctl mask apt timer out: %s", output)
+        output = subprocess.check_output(
+            ["/usr/bin/systemctl", "stop", APT_UPGRADE_TIMER], timeout=30
+        )  # nosec: B603
+        logger.info("systemctl stop apt upgrade timer out: %s", output)
+        output = subprocess.check_output(  # nosec: B603
+            ["/usr/bin/systemctl", "disable", APT_UPGRADE_TIMER], timeout=30
+        )
+        logger.info("systemctl disable apt upgrade timer out: %s", output)
+        output = subprocess.check_output(
+            ["/usr/bin/systemctl", "mask", APT_UPGRAD_SVC], timeout=30
+        )  # nosec: B603
+        logger.info("systemctl mask apt upgrade timer out: %s", output)
+        output = subprocess.check_output(
+            ["/usr/bin/systemctl", "daemon-reload"], timeout=30
+        )  # nosec: B603
+        logger.info("systemctl daemon-reload out: %s", output)
+        output = subprocess.check_output(  # nosec: B603
+            ["/usr/bin/apt-get", "remove", "-y", "unattended-upgrades"], timeout=30
+        )
+        logger.info("apt-get remove unattended-upgrades out: %s", output)
+    except subprocess.CalledProcessError as exc:
+        logger.exception(
+            "Error disabling unattended upgrades, cmd: %s, code: %s, err: %s",
+            exc.cmd,
+            exc.returncode,
+            exc.output,
+        )
+        raise UnattendedUpgradeDisableError from exc
+
+
+def _configure_system_users() -> None:
+    """Configure system users.
+
+    Raises:
+        SystemUserConfigurationError: If there was an error configuring ubuntu user.
+    """
+    try:
+        output = subprocess.check_output(  # nosec: B603
+            ["/usr/sbin/useradd", "-m", UBUNTU_USER], timeout=30
+        )
+        logger.info("useradd ubunutu out: %s", output)
+        with (UBUNTU_HOME / ".profile").open("a") as profile_file:
+            profile_file.write(f"PATH=$PATH:{UBUNTU_HOME}/.local/bin\n")
+        with (UBUNTU_HOME / ".bashrc").open("a") as bashrc_file:
+            bashrc_file.write(f"PATH=$PATH:{UBUNTU_HOME}/.local/bin\n")
+        output = subprocess.check_output(
+            ["/usr/sbin/groupadd", MICROK8S_GROUP], timeout=30
+        )  # nosec: B603
+        logger.info("groupadd microk8s out: %s", output)
+        output = subprocess.check_output(  # nosec: B603
+            ["/usr/sbin/usermod", "-aG", DOCKER_GROUP, UBUNTU_USER], timeout=30
+        )
+        logger.info("groupadd add ubuntu to docker group out: %s", output)
+        output = subprocess.check_output(  # nosec: B603
+            ["/usr/sbin/usermod", "-aG", MICROK8S_GROUP, UBUNTU_USER], timeout=30
+        )
+        logger.info("groupadd add ubuntu to microk8s group out: %s", output)
+        output = subprocess.check_output(  # nosec: B603
+            ["/usr/sbin/usermod", "-aG", LXD_GROUP, UBUNTU_USER], timeout=30
+        )
```

Comment:
`-G` in `usermod` accepts a list of groups to add (comma separated).
This can be done in a single command.